### ISSUE TITLE
docs: Use relative paths in docstrings

### ISSUE THREAD

**Issue by MarcoGorelli** [narwhals-dev/narwhals#1629]
For reference: https://github.com/narwhals-dev/narwhals/pull/1624#pullrequestreview-2517253746

Relative paths still show up correctly in the official docs, but also show up when building docs locally

**Comment by DeaMariaLeon** [comment]
@MarcoGorelli when using links like `[`narwhals.from_dict`](../narwhals/#narwhals.from_native)`  warnings are generated when building the docs locally and the CI:

```
INFO    -  Doc file 'api-reference/dataframe.md' contains an unrecognized relative link '../narwhals/#narwhals.from_native', it was left as is. Did you mean 'narwhals.md#narwhals.from_native'?
INFO    -  Doc file 'api-reference/dataframe.md' contains an unrecognized relative link '../narwhals/#narwhals.from_dict', it was left as is. Did you mean 'narwhals.md#narwhals.from_dict'?
...
```

I think we should write like this ``[`narwhals.from_native`][]`` instead. That way the are gone (and they work).

**Comment by MarcoGorelli** [comment]
sure, thanks!

**Comment by FBruzzesi** [comment]
Closed by #1631

**Comment by DeaMariaLeon** [comment]
Yes @FBruzzesi, I think I should still change some links. I'm looking at that now. 😇
